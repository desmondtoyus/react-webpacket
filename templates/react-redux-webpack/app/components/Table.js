import React, { Component } from 'react';
import PropTypes from 'prop-types';


class Table extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    const { content, label } = this.props;
    return (
      <main>
        {content.length ? (
          <div>
            <span> {label} </span>
            <table className="table  table-sm">
              <thead className="table-info">
                <tr>
                  <th />
                  <th>Name</th>
                  <th>Registration Date</th>
                  <th>E-mail address</th>
                  <th>Premium Plan</th>
                </tr>
              </thead>
              <tbody>
                {content.map(item => (
                  <tr key={item.id}>
                    <td>
                      <label htmlFor="checkbox" className="custom-control custom-checkbox">
                        <input type="checkbox" className="custom-control-input" />
                        <span className="custom-control-indicator" />
                      </label>
                    </td>
                    <td>{item.name}</td>
                    <td>{item.date}</td>
                    <td>{item.email}</td>
                    <td>{item.plan}</td>
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        ) : <span> No Content</span>}
      </main>
    );
  }
}

// Do not include required props in the defaultProps
Table.propTypes = {
  label: PropTypes.string.isRequired,
  content: PropTypes.arrayOf(PropTypes.object),
};

Table.defaultProps = {
  content: [],
};

export default Table;
