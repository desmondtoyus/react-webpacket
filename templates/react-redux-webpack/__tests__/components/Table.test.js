import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
// import toJson from 'enzyme-to-json';
// import configureStore from 'redux-mock-store'
// The component folder shoould hold unit test(component) using shallow, 
// while view folder does intergration test with 'mount'
import Table from "../../app/components/Table";
let  mockContent =[
    {name: 'Desmondjhjhhj', date:'Nov 4 1998', email:'desmondkjjkjh@pilotx.tv', plan:'opoCustom'},
    {name: 'Desmond2,klk', date:'Nov 4 1955', email:'desmond2l;;l@pilotx.tv', plan:'kjpaid'},
    {name: 'Desmondjhjhhj', date:'Nov 4 1998', email:'desmondkjjkjh@pilotx.tv', plan:'opoCustom'},
    {name: 'Desmond2,klk', date:'Nov 4 1955', email:'desmond2l;;l@pilotx.tv', plan:'kjpaid'}
]

describe('Table Component', () => {
        test('renders correctly the table component', () => {
            const wrapper = shallow(<Table label={'Test label'} />);
            expect (wrapper.exists()).toBeTruthy();
          });

      it('renders Table component', () => {
        const wrapper = renderer.create(<Table label={'Test Table'} content={mockContent}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
      });
  

    it('renders text in the right possition', () => {
      const wrapper = mount(<Table label={'Test label'} content={mockContent}/>);
      //Expect the child of the first item to be an array
      expect(wrapper.find('td').get(1).props.children).toEqual('Desmondjhjhhj');
    });

    it('render table correctly with empty props', ()=>{
      mockContent =[];
        const wrapper = mount(<Table label={'Test label'} content={mockContent}/>);
        expect((wrapper).prop('content')).toHaveLength(0);
    });

    // Testing  whether component rerenders when the state changes
    test('rerenders when the state changes', ()=>{
      mockContent=[{name: 'Ademiluyi', date:'Dec 30 1996', email:'OlaifeOluwa@gmail.com', plan:'Full'}]
        const wrapper = shallow(<Table label={'Test label'} content={mockContent} />);
        wrapper.setProps({
            content:[{name: 'Desmond', date:'Dec 30 1996', email:'OlaifeOluwa@gmail.com', plan:'Full'}]
        })
        console.log(wrapper);
        expect(wrapper.contains(<td>Desmond</td>)).toBeTruthy();
    });
});