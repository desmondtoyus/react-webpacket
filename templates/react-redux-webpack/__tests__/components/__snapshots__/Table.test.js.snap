// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component renders Table component 1`] = `
<main>
  <div>
    <span>
       
      Test Table
       
    </span>
    <table
      className="table  table-sm"
    >
      <thead
        className="table-info"
      >
        <tr>
          <th />
          <th>
            Name
          </th>
          <th>
            Registration Date
          </th>
          <th>
            E-mail address
          </th>
          <th>
            Premium Plan
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <label
              className="custom-control custom-checkbox"
            >
              <input
                className="custom-control-input"
                type="checkbox"
              />
              <span
                className="custom-control-indicator"
              />
            </label>
          </td>
          <td>
            Desmondjhjhhj
          </td>
          <td>
            Nov 4 1998
          </td>
          <td>
            desmondkjjkjh@pilotx.tv
          </td>
          <td>
            opoCustom
          </td>
        </tr>
        <tr>
          <td>
            <label
              className="custom-control custom-checkbox"
            >
              <input
                className="custom-control-input"
                type="checkbox"
              />
              <span
                className="custom-control-indicator"
              />
            </label>
          </td>
          <td>
            Desmond2,klk
          </td>
          <td>
            Nov 4 1955
          </td>
          <td>
            desmond2l;;l@pilotx.tv
          </td>
          <td>
            kjpaid
          </td>
        </tr>
        <tr>
          <td>
            <label
              className="custom-control custom-checkbox"
            >
              <input
                className="custom-control-input"
                type="checkbox"
              />
              <span
                className="custom-control-indicator"
              />
            </label>
          </td>
          <td>
            Desmondjhjhhj
          </td>
          <td>
            Nov 4 1998
          </td>
          <td>
            desmondkjjkjh@pilotx.tv
          </td>
          <td>
            opoCustom
          </td>
        </tr>
        <tr>
          <td>
            <label
              className="custom-control custom-checkbox"
            >
              <input
                className="custom-control-input"
                type="checkbox"
              />
              <span
                className="custom-control-indicator"
              />
            </label>
          </td>
          <td>
            Desmond2,klk
          </td>
          <td>
            Nov 4 1955
          </td>
          <td>
            desmond2l;;l@pilotx.tv
          </td>
          <td>
            kjpaid
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</main>
`;
